qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AttachmentsPage.java

```diff
@@ -34,6 +41,22 @@ public void setFileUrl(String fileUrl) {
 		WebElement element = driver.findElement(By.xpath("//input[@type='file']"));
 		element.sendKeys(fileUrl);
 	}
+	
+	public String createRandomFile() throws IOException {
+		File tempFile = File.createTempFile("file", ".pdf");
+		tempFile.deleteOnExit();
+		try (InputStream pdf = getClass().getClassLoader().getResourceAsStream("data/file.pdf")) {
+			try (Writer writer = new FileWriter(tempFile)) {
+				IOUtils.copy(pdf, writer, StandardCharsets.UTF_8);
+			}
+		}
+		return tempFile.getAbsolutePath();
+	}
+	
+	public void getRandomFile() throws IOException {
```

Comment:
Do we need this method? It seems like we could replace any use of this with `setFileUrl(createRandomFile())` (and if we do need this method, that should be body.

Reply:
@ibacher we wanted to leave the setFileUrl() to handle the hard coded file path just in case it is required. Do you think it is not necessary ?

Reply:
> Do we need this method? It seems like we could replace any use of this with `setFileUrl(createRandomFile())` (and if we do need this method, that should be body.

@ibacher did you imply that instead of having this as a method, it would be better to instead have the code as the body of setFileUrl()? I hope I understood your suggestion.